docs/operations/clean-metadata-store.md

```diff
@@ -0,0 +1,104 @@
+
+## Automated cleanup strategies
+There are several cases when you should consider automated cleanup of the metadata related to deleted datasources:
+- Proactively, if you know you have many high-churn datasources. For example you have scripts that create and delete supervisors regularly.
+- If you have issues with the hard disk for your metadata database filling up.
+- If you run into performance issues with the metadata database. For example API calls are very slow or fail to execute.
+
+Do not use the metadata store automated cleanup features if you have requirements to retain metadata records. For example, you have compliance requirements to keep audit records. In these cases, you should come up with an alternate method to preserve the audit metadata while freeing up your active metadata store.
+
+## Configure automated metadata cleanup
+Automated cleanup only removes records for deleted datasources. You can configure cleanup on a per-table basis as follows:
+ - `druid.coordinator.kill.*.on` enables cleanup for a particular metadata table.
+ - `druid.coordinator.kill.*.period` defines the frequency in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) for the cleanup job to check for and delete eligible records.
+ - `druid.coordinator.kill.*.durationToRetain` defines the retention period in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) after creation that the entity becomes eligible for deletion. For example to retain records for 30 days from their creation date: `P30D`.
+
+ For full configuration details, see [Metadata management](../configuration/index.md#metadata-management).
+
+The following are eligible for automatic cleanup:
+- `druid_audit`: Audit records for deleted datasources become eligible for deletion when the `durationToRetain` time has passed since their creation.
+- `druid_supervisors`: Supervisor records become eligible for deletion when:
+
+    - the supervisor is terminated.
+    - the `durationToRetain` time has passed since their creation.
+- `druid_rules`: Rule records become eligible for deletion when:
+
+    - no segments exist for the datasource.
+    - the `durationToRetain` time has passed since their creation.
+- `compaction`: Compaction configuration records in the records in the `druid_config` table become eligible for deletion when no segments exist for the datasource.
+     >If you already have an extremely large compaction configuration, you may not be able to delete compaction configuration due to size limits with the audit log. In this case you can set `druid.audit.manager.maxPayloadSizeBytes` and `druid.audit.manager.skipNullField` to avoid the auditing issue. See [Audit logging](../configuration/index.md#audit-logging).
+- `druid_datasources`: Datasource records become eligible for deletion when:
```

Comment:
Might be worth to note that druid_datasources entry only exists for datasource with a supervisor. (the namely of the table is misleading as not all datasource has entry in the datasources table